osquery/dispatcher/scheduled_tasks_queue.h

```diff
@@ -0,0 +1,78 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#pragma once
+
+#include <functional>
+#include <vector>
+
+#include <osquery/system.h>
+
+namespace osquery {
+
+class ScheduledTaskQueue {
```

Comment:
I love the use of a heap here! My only question is that it's implemented as a Priority Queue but is not called as such, instead being called a queue, which it isn't.

What do you think of changing the name to one of: Heap, PQueue, PriorityQueue?

Reply:
Thank you.

My thought was about _schedule_. If I have something to do in real life, I put it in the _schedule_ in my calendar with a time when to do it. And If I want to know what to do I go to my schedule and look if I have something to do now. If there is something - I grab it and do, if no - I wait some time.

There is nothing about priority or about heap, it is a details of implementation, which is good to hide from user under the interface. All the actions is about schedule or calendar. So thats why I use it in the name of Class.

We can change it to the `TaskSchedule` or `TaskScheduler` or something similar. What do you think?

Reply:
I like `TaskSchedule`, done